tools/interface/ontology.py

```diff
@@ -0,0 +1,61 @@
+import sys
+from typing import List, NamedTuple
+
+sys.path.insert(1, '../validators/ontology_validator/')
+
+from yamlformat.validator.entity_type_lib import EntityType
+from yamlformat.validator.presubmit_validate_types_lib import ConfigUniverse
+from yamlformat import constants
+
+StandardField = NamedTuple('StandardField', [('namespace', str), ('name', str), ('is_required', bool)])
+
+StandardSubfield = NamedTuple('StandardSubfield', [('namespace', str), ('name', str)])
+
+class Ontology(object):
+    """Class defining an implementation of Google's ConfigUniverse as an ontology
+
+       Attributes:
+            self.universe = A ConfigUniverse object detailing the various universes in the ontology
+    """
+
+    def __init__(self, universe: ConfigUniverse):
+        super().__init__()
+        self.universe = universe
+
+    def GetFieldsForTypeName(self, entity_type: EntityType, required_only: bool = False) -> List[StandardField]:
+        """gets a list of fields for a given typename within a namespace
+
+           Args:
+                entity_type: An EntityType object to get fields for 
+                required_only: when true will return only required fields for a given type
+
+           Returns:
+                result_fields: a list of StandardField tuples
+        """
+        pass
+     
+    def GetEntitiesFromFields(self, field_list: List[StandardField], general_type: str = None) -> List[EntityType]:
+        """gets a list of EntityType objects matching a given a list of StandardField tuples
+
+           Args:
+                field_list: a list of StandardField tuples to match to an entity
+                general_type: a string indicating a general type name to narrow return results
+
+            Returns:
+                entities: a list of EntityType objects matching the provided list of fields
+        """
+        pass
+
+    def IsTypeValid(self, entity_type: EntityType) -> bool:
+        """Validates the existence of a type in the ontology"""
+        pass
```

Comment:
If you already have an EntityType object, can't you just call IsValid() on it?  Maybe I'm missing something.